src/index.js

```diff
@@ -1,42 +1,94 @@
-    li.innerHTML = `
-      <div class="list_con">
-        <input type="checkbox" ${checkcompleted}>
-        <p>${list[i].description}</p>
-      </div>
-                  
-      <button class="listBtn"><i class="fa-solid fa-ellipsis-vertical"></i></button>
-    
-    `;
-    listCont.appendChild(li);
+function createTodoElement(todo) {
+  const todoElement = document.createElement('li');
+  todoElement.classList.add('list_item');
+  todoElement.innerHTML = `
+    <div class='list_con'>
+      <input class="input1" type="checkbox" ${todo.completed ? 'checked' : ''}>
+      <span class='p' contenteditable="true" class="${todo.completed ? 'completed' : ''}">${todo.task}</span>    
+    </div>
+    <div>
+      <button class="btns edit-todo"><i class="fa-solid fa-ellipsis-vertical"></i></button>
+      <button class="btns remove-todo" style="display: none;"><i class="fa-solid fa-trash-can"></i></button>
+      <button class="btns save-todo" style="display: none;">Save</button>
+    </div>   
+  `;
+  const todoTextElement = todoElement.querySelector('span');
+  const editTodoButton = todoElement.querySelector('.edit-todo');
+  const removeTodoButton = todoElement.querySelector('.remove-todo');
+  const saveTodoButton = todoElement.querySelector('.save-todo');
+  todoTextElement.addEventListener('keydown', (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      saveTodoButton.click();
+    }
+  });
+  saveTodoButton.addEventListener('click', () => {
+    todo.task = todoTextElement.textContent.trim();
+    saveTodos(todos);
+    saveTodoButton.style.display = 'none';
+  });
+  todoElement.querySelector('input').addEventListener('change', () => {
+    todo.completed = !todo.completed;
+    saveTodos(todos);
+    if (todo.completed) {
+      todoTextElement.classList.add('completed');
+    } else {
+      todoTextElement.classList.remove('completed');
+    }
+  });
+  editTodoButton.addEventListener('click', () => {
+    todoTextElement.contentEditable = true;
+    todoElement.style.background = 'lightyellow';
+    editTodoButton.style.display = 'none';
+    removeTodoButton.style.display = 'inline-block';
+  });
+  todoElement.querySelector('button.remove-todo').addEventListener('click', () => {
+    const index = todos.indexOf(todo);
+    todos.splice(index, 1);
+    saveTodos(todos);
+    todos.forEach((todo, index) => {
+      todo.index = index;
+    });
+    todosList.innerHTML = '';
+    todos.forEach((todo, index) => {
+      todo.index = index;
+      todosList.appendChild(createTodoElement(todo));
+    });
+  });
+  return todoElement;
+}
+function renderTodos() {
+  todosList.innerHTML = '';
+  todos.forEach((todo, index) => {
+    todo.index = index;
+    todosList.appendChild(createTodoElement(todo));
+  });
+}
+addTodoBtn.addEventListener('click', () => {
+  const task = todoInput.value.trim();
+  if (task) {
+    todos.push({ task, completed: false, index: todos.length });
+    saveTodos(todos);
```

Comment:
- [ ] According to requirements, the property `index` should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). This means your indices would start at 1 instead of 0 👍 

![image](https://user-images.githubusercontent.com/46839548/222865142-bac20fed-572c-4afb-b3ca-5bd5cd0637db.png)